src/aks-preview/azext_aks_preview/_validators.py

```diff
@@ -504,3 +504,16 @@ def validate_assign_kubelet_identity(namespace):
         from msrestazure.tools import is_valid_resource_id
         if not is_valid_resource_id(namespace.assign_kubelet_identity):
             raise CLIError("--assign-kubelet-identity is not a valid Azure resource ID.")
+
+
+def validate_nodepool_id(namespace):
+    from msrestazure.tools import is_valid_resource_id
+    if not is_valid_resource_id(namespace.nodepool_id):
+        raise CLIError("--nodepool-id is not a valid Azure resource ID.")
```

Comment:
Could you please use the specific error type instead of `CLIError`?

Reply:
Change it to InvalidArgumentValueError, although in existing code, the CLIError are used many times.